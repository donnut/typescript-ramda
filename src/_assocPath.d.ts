/**
 * This is auto generated source.
 */

import { Path } from '../interfaces'

interface assocPath {
  /**
   * 1Returns a new function much like the supplied one, except that the first two arguments'
   * order is reversed.
   */  
  <T, U>(path: Path, val: T, obj: U): U;
  /**
   * 2Returns a new function much like the supplied one, except that the first two arguments'
   * order is reversed.
   */    
  <T>(path: Path, val: T): {
    <U>(obj: U): U;
  };
  /**
   * 3Returns a new function much like the supplied one, except that the first two arguments'
   * order is reversed.
   */  

  (path: Path): {
    /**
     * 4Returns a new function much like the supplied one, except that the first two arguments'
     * order is reversed.
     */  
    <T, U>(val: T, obj: U): U;
    <T>(val: T): {
      <U>(obj: U): U;
    };
  };
}
declare const x: assocPath

export = x
