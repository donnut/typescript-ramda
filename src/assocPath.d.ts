/**
 * This is auto generated source.
 */

import { Path, List, Prop } from '../interfaces';


/**
 * Makes a shallow clone of an object, setting or overriding 
 * the nodes required to create the given path, and placing the specific 
 * value at the tail end of that path.
 * 
 * Note: Tuples are not supported for typings.
 */
declare function assocPath<P1 extends string | number,
  P2 extends string | number, P3 extends string | number,
  P4 extends string | number, P5 extends string | number,
  P6 extends string | number, P7 extends string | number,
  P8 extends string | number, T,
  O extends {[K1 in P1]: {[K2 in P2]: {[K3 in P3]: {[K4 in P4]: {[K5 in P5]: {[K6 in P6]: {[K7 in P7]: {[K8 in P8]: T}}}}}}}}>(
  path: [P1, P2, P3, P4, P5, P6, P7, P8], value: T, obj: O): O;


/**
 * Makes a shallow clone of an object, setting or overriding 
 * the nodes required to create the given path, and placing the specific 
 * value at the tail end of that path.
 * 
 * Note: Tuples are not supported for typings.
 */
declare function assocPath<P1 extends string | number,
  P2 extends string | number, P3 extends string | number,
  P4 extends string | number, P5 extends string | number,
  P6 extends string | number, P7 extends string | number, T,
  O extends {[K1 in P1]: {[K2 in P2]: {[K3 in P3]: {[K4 in P4]: {[K5 in P5]: {[K6 in P6]: {[K7 in P7]: T}}}}}}}>(
  path: [P1, P2, P3, P4, P5, P6, P7], value: T, obj: O): O;


/**
 * Makes a shallow clone of an object, setting or overriding 
 * the nodes required to create the given path, and placing the specific 
 * value at the tail end of that path.
 * 
 * Note: Tuples are not supported for typings.
 */
declare function assocPath<P1 extends string | number,
  P2 extends string | number, P3 extends string | number,
  P4 extends string | number, P5 extends string | number,
  P6 extends string | number, T,
  O extends {[K1 in P1]: {[K2 in P2]: {[K3 in P3]: {[K4 in P4]: {[K5 in P5]: {[K6 in P6]: T}}}}}}>(
  path: [P1, P2, P3, P4, P5, P6], value: T, obj: O): O;


/**
 * Makes a shallow clone of an object, setting or overriding 
 * the nodes required to create the given path, and placing the specific 
 * value at the tail end of that path.
 * 
 * Note: Tuples are not supported for typings.
 */
declare function assocPath<P1 extends string | number,
  P2 extends string | number, P3 extends string | number,
  P4 extends string | number, P5 extends string | number, T,
  O extends {[K1 in P1]: {[K2 in P2]: {[K3 in P3]: {[K4 in P4]: {[K5 in P5]: T}}}}}>(
  path: [P1, P2, P3, P4, P5], value: T, obj: O): O;


/**
 * Makes a shallow clone of an object, setting or overriding 
 * the nodes required to create the given path, and placing the specific 
 * value at the tail end of that path.
 * 
 * Note: Tuples are not supported for typings.
 */
declare function assocPath<P1 extends string | number,
  P2 extends string | number, P3 extends string | number,
  P4 extends string | number, T,
  O extends {[K1 in P1]: {[K2 in P2]: {[K3 in P3]: {[K4 in P4]: T}}}}>(
  path: [P1, P2, P3, P4], value: T, obj: O): O;


/**
 * Makes a shallow clone of an object, setting or overriding 
 * the nodes required to create the given path, and placing the specific 
 * value at the tail end of that path.
 * 
 * Note: Tuples are not supported for typings.
 */
declare function assocPath<P1 extends string | number,
  P2 extends string | number, P3 extends string | number, T,
  O extends {[K1 in P1]: {[K2 in P2]: {[K3 in P3]: T}}}>(
  path: [P1, P2, P3], value: T, obj: O): O;


/**
 * Makes a shallow clone of an object, setting or overriding 
 * the nodes required to create the given path, and placing the specific 
 * value at the tail end of that path.
 * 
 * Note: Tuples are not supported for typings.
 */
declare function assocPath<P1 extends string | number,
  P2 extends string | number, T, O extends {[K1 in P1]: {[K2 in P2]: T}}>(
  path: [P1, P2], value: T, obj: O): O;


/**
 * Makes a shallow clone of an object, setting or overriding 
 * the nodes required to create the given path, and placing the specific 
 * value at the tail end of that path.
 * 
 * Note: Tuples are not supported for typings.
 */
declare function assocPath<P1 extends string | number, T,
  O extends {[K1 in P1]: T}>(path: [P1], value: T, obj: O): O;

// curried


/**
 * Makes a shallow clone of an object, setting or overriding 
 * the nodes required to create the given path, and placing the specific 
 * value at the tail end of that path.
 * 
 * Note: Tuples are not supported for typings.
 */
declare function assocPath<P1 extends string, P2 extends string,
  P3 extends string, P4 extends string, P5 extends string,
  P6 extends string, P7 extends string, P8 extends string>(
  ps: [P1, P2, P3, P4, P5, P6, P7, P8]):
  <O extends {[K in P1 | P2 | P3 | P4 | P5 | P6 | P7 | P8]: any}>(obj: O) =>
  [O[P1], O[P2], O[P3], O[P4], O[P5], O[P6], O[P7], O[P8]];


/**
 * Makes a shallow clone of an object, setting or overriding 
 * the nodes required to create the given path, and placing the specific 
 * value at the tail end of that path.
 * 
 * Note: Tuples are not supported for typings.
 */
declare function assocPath<P1 extends string, P2 extends string,
  P3 extends string, P4 extends string, P5 extends string,
  P6 extends string, P7 extends string>(ps: [P1, P2, P3, P4, P5, P6, P7]):
  <O extends {[K in P1 | P2 | P3 | P4 | P5 | P6 | P7]: any}>(obj: O) =>
  [O[P1], O[P2], O[P3], O[P4], O[P5], O[P6], O[P7]];


/**
 * Makes a shallow clone of an object, setting or overriding 
 * the nodes required to create the given path, and placing the specific 
 * value at the tail end of that path.
 * 
 * Note: Tuples are not supported for typings.
 */
declare function assocPath<P1 extends string, P2 extends string,
  P3 extends string, P4 extends string, P5 extends string, P6 extends string>(
  ps: [P1, P2, P3, P4, P5, P6]):
  <O extends {[K in P1 | P2 | P3 | P4 | P5 | P6]: any}>(obj: O) =>
  [O[P1], O[P2], O[P3], O[P4], O[P5], O[P6]];


/**
 * Makes a shallow clone of an object, setting or overriding 
 * the nodes required to create the given path, and placing the specific 
 * value at the tail end of that path.
 * 
 * Note: Tuples are not supported for typings.
 */
declare function assocPath<P1 extends string, P2 extends string,
  P3 extends string, P4 extends string, P5 extends string>(
  ps: [P1, P2, P3, P4, P5]):
  <O extends {[K in P1 | P2 | P3 | P4 | P5]: any}>(obj: O) =>
  [O[P1], O[P2], O[P3], O[P4], O[P5]];


/**
 * Makes a shallow clone of an object, setting or overriding 
 * the nodes required to create the given path, and placing the specific 
 * value at the tail end of that path.
 * 
 * Note: Tuples are not supported for typings.
 */
declare function assocPath<P1 extends string, P2 extends string,
  P3 extends string, P4 extends string>(ps: [P1, P2, P3, P4]):
  <O extends {[K in P1 | P2 | P3 | P4]: any}>(obj: O) =>
  [O[P1], O[P2], O[P3], O[P4]];


/**
 * Makes a shallow clone of an object, setting or overriding 
 * the nodes required to create the given path, and placing the specific 
 * value at the tail end of that path.
 * 
 * Note: Tuples are not supported for typings.
 */
declare function assocPath<P1 extends string, P2 extends string,
  P3 extends string>(ps: [P1, P2, P3]):
  <O extends {[K in P1 | P2 | P3]: any}>(obj: O) => [O[P1], O[P2], O[P3]];


/**
 * Makes a shallow clone of an object, setting or overriding 
 * the nodes required to create the given path, and placing the specific 
 * value at the tail end of that path.
 * 
 * Note: Tuples are not supported for typings.
 */
declare function assocPath<P1 extends string, P2 extends string>(
  ps: [P1, P2]):
  <O extends {[K in P1 | P2]: any}>(obj: O) => [O[P1], O[P2]];


/**
 * Makes a shallow clone of an object, setting or overriding 
 * the nodes required to create the given path, and placing the specific 
 * value at the tail end of that path.
 * 
 * Note: Tuples are not supported for typings.
 */
declare function assocPath<P1 extends string>(ps: [P1]):
  <O extends {[K in P1]: any}>(obj: O) => [O[P1]];


/**
 * Makes a shallow clone of an object, setting or overriding 
 * the nodes required to create the given path, and placing the specific 
 * value at the tail end of that path.
 * 
 * Note: Tuples are not supported for typings.
 */
declare function props<T>(ps: List<Prop>, obj: List<T>): T[];

// curried


/**
 * Makes a shallow clone of an object, setting or overriding 
 * the nodes required to create the given path, and placing the specific 
 * value at the tail end of that path.
 * 
 * Note: Tuples are not supported for typings.
 */
declare function props<T>(ps: List<Prop>): <T>(obj: List<T>) => T[];

export = assocPath
