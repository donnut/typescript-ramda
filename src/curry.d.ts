
type CurriedFunction1<T1, R> = (v1: T1) => R;

interface CurriedFunction2<T1, T2, R> {
  (v1: T1): (v2: T2) => R;
  (v1: T1, v2: T2): R;
}

interface CurriedFunction3<T1, T2, T3, R> {
  (v1: T1): CurriedFunction2<T2, T3, R>;
  (v1: T1, v2: T2): (v3: T3) => R;
  (v1: T1, v2: T2, v3: T3): R;
}

interface CurriedFunction4<T1, T2, T3, T4, R> {
  (v1: T1): CurriedFunction3<T2, T3, T4, R>;
  (v1: T1, v2: T2): CurriedFunction2<T3, T4, R>;
  (v1: T1, v2: T2, v3: T3): (v4: T4) => R;
  (v1: T1, v2: T2, v3: T3, v4: T4): R;
}

interface CurriedFunction5<T1, T2, T3, T4, T5, R> {
  (v1: T1): CurriedFunction4<T2, T3, T4, T5, R>;
  (v1: T1, v2: T2): CurriedFunction3<T3, T4, T5, R>;
  (v1: T1, v2: T2, v3: T3): CurriedFunction2<T4, T5, R>;
  (v1: T1, v2: T2, v3: T3, v4: T4): (v5: T5) => R;
  (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5): R;
}

interface CurriedFunction6<T1, T2, T3, T4, T5, T6, R> {
  (v1: T1): CurriedFunction5<T2, T3, T4, T5, T6, R>;
  (v1: T1, v2: T2): CurriedFunction4<T3, T4, T5, T6, R>;
  (v1: T1, v2: T2, v3: T3): CurriedFunction3<T4, T5, T6, R>;
  (v1: T1, v2: T2, v3: T3, v4: T4): CurriedFunction2<T5, T6, R>;
  (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5): (v6: T6) => R;
  (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5, v6: T6): R;
}

interface CurriedFunction7<T1, T2, T3, T4, T5, T6, T7, R> {
  (v1: T1): CurriedFunction6<T2, T3, T4, T5, T6, T7, R>;
  (v1: T1, v2: T2): CurriedFunction5<T3, T4, T5, T6, T7, R>;
  (v1: T1, v2: T2, v3: T3): CurriedFunction4<T4, T5, T6, T7, R>;
  (v1: T1, v2: T2, v3: T3, v4: T4): CurriedFunction3<T5, T6, T7, R>;
  (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5): CurriedFunction2<T6, T7, R>;
  (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5, v6: T6): (v7: T7) => R;
  (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5, v6: T6, v7: T7): R;
}

interface CurriedFunction8<T1, T2, T3, T4, T5, T6, T7, T8, R> {
  (v1: T1): CurriedFunction7<T2, T3, T4, T5, T6, T7, T8, R>;
  (v1: T1, v2: T2): CurriedFunction6<T3, T4, T5, T6, T7, T8, R>;
  (v1: T1, v2: T2, v3: T3): CurriedFunction5<T4, T5, T6, T7, T8, R>;
  (v1: T1, v2: T2, v3: T3, v4: T4): CurriedFunction4<T5, T6, T7, T8, R>;
  (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5): CurriedFunction3<T6, T7, T8, R>;
  (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5, v6: T6): CurriedFunction2<T7, T8, R>;
  (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5, v6: T6, v7: T7): (v8: T8) => R;
  (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5, v6: T6, v7: T7, v8: T8): R;
}

interface CurriedFunction9<T1, T2, T3, T4, T5, T6, T7, T8, T9, R> {
  (v1: T1): CurriedFunction8<T2, T3, T4, T5, T6, T7, T8, T9, R>;
  (v1: T1, v2: T2): CurriedFunction7<T3, T4, T5, T6, T7, T8, T9, R>;
  (v1: T1, v2: T2, v3: T3): CurriedFunction6<T4, T5, T6, T7, T8, T9, R>;
  (v1: T1, v2: T2, v3: T3, v4: T4): CurriedFunction5<T5, T6, T7, T8, T9, R>;
  (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5): CurriedFunction4<T6, T7, T8, T9, R>;
  (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5, v6: T6): CurriedFunction3<T7, T8, T9, R>;
  (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5, v6: T6, v7: T7): CurriedFunction2<T8, T9, R>;
  (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5, v6: T6, v7: T7, v8: T8): (v9: T9) => R;
  (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5, v6: T6, v7: T7, v8: T8, v9: T9): R;
}

/**
 * Returns a curried equivalent of the provided function.
 */
declare function curry<T1, TResult>(fn: (a: T1) => TResult): CurriedFunction1<T1, TResult>;
declare function curry<T1, T2, TResult>(fn: (a: T1, b: T2) => TResult): CurriedFunction2<T1, T2, TResult>;
declare function curry<T1, T2, T3, T4, TResult>(fn: (a: T1, b: T2, c: T3, d: T4) => TResult): CurriedFunction4<T1, T2, T3, T4, TResult>;
declare function curry<T1, T2, T3, T4, T5, TResult>(fn: (a: T1, b: T2, c: T3, d: T4, e: T5) => TResult): CurriedFunction5<T1, T2, T3, T4, T5, TResult>;
declare function curry<T1, T2, T3, T4, T5, T6, TResult>(fn: (a: T1, b: T2, c: T3, d: T4, e: T5, f: T6) => TResult): CurriedFunction6<T1, T2, T3, T4, T5, T6, TResult>;
declare function curry<T1, T2, T3, T4, T5, T6, T7, TResult>(fn: (a: T1, b: T2, c: T3, d: T4, e: T5, f: T6, g: T7) => TResult): CurriedFunction7<T1, T2, T3, T4, T5, T6, T7, TResult>;
declare function curry<T1, T2, T3, T4, T5, T6, T7, T8, TResult>(fn: (a: T1, b: T2, c: T3, d: T4, e: T5, f: T6, g: T7, h: T8) => TResult): CurriedFunction8<T1, T2, T3, T4, T5, T6, T7, T8, TResult>;
declare function curry<T1, T2, T3, T4, T5, T6, T7, T8, T9, TResult>(fn: (a: T1, b: T2, c: T3, d: T4, e: T5, f: T6, g: T7, h: T8, i: T9) => TResult): CurriedFunction9<T1, T2, T3, T4, T5, T6, T7, T8, T9, TResult>;

export default curry;
