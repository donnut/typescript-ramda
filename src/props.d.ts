import { Struct, List, Prop } from '../interfaces'

/**
 * Acts as multiple `prop`: array of keys in, array of values out. Preserves order.
 */
declare function props<
  P1 extends string, P2 extends string,
  O extends {[K in P1 | P2]: any}>
  (ps: [P1, P2], obj: O): [O[P1], O[P2]];

declare function props<
  P1 extends string,
  O extends {[K in P1]: any}>
  (ps: [P1], obj: O): [O[P1]];


declare function props<
  P1 extends string, P2 extends string>
  (ps: [P1, P2]): <O extends {[K in P1 | P2]: any}>(obj: O) => [O[P1], O[P2]];


declare function props<P1 extends string>
  (ps: [P1]): <O extends {[K in P1]: any}>(obj: O) => [O[P1]];

// declare function props<P1 extends string, P2 extends string>
//   (ps: [P1, P2]): <T1, T2>(obj: {[K in P1]: T1} & {[K2 in P2]: T2})
//     => [T1, T2];


//declare function props<T>(ps: List<Prop>, obj: List<T>): T[];
//declare function props(ps: List<Prop>): <T>(obj: List<T>) => T[];

export = props;
