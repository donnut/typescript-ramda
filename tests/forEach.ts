import { List, Tap } from '../ramda/dist/src/$types';
import R_forEach = require('../ramda/dist/src/forEach');

declare const number_list: List<number>;
declare const number_tap: Tap<number>;

// @dts-jest:pass:snap
R_forEach(number_tap);
// @dts-jest:pass:snap
R_forEach(number_tap)(number_list);
// @dts-jest:pass:snap
R_forEach(number_tap, number_list);
