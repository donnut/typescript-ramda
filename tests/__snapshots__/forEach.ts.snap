// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`R.forEach(number_tap) 1`] = `"<U extends number[] | ArrayLike<number>>(list: U) => U"`;

exports[`R.forEach(number_tap)(number_list) 1`] = `"number[] | ArrayLike<number>"`;

exports[`R.forEach(number_tap, number_list) 1`] = `"number[] | ArrayLike<number>"`;
